### PR TITLE
Pair corr estimator norm factor fix

### PR DESCRIPTION
**Pull request by jptowns**
Fixed PairCorrEstimator so that it will pass unit test.

The reason it failed was that norm_factor was built using two approximations:
1.) Small delta approximation: (r+delta)^3 - r^3 = 3*delta*r^2 + 3*r*delta^2 ~ 3*delta*r^2
2.) Large N approximation: N(N-1) = N^2 - N ~ N^2

Both approximations are badly violated for a small cell with a large delta and small number of particles, which is exactly the case for the unit test.

Because computing the normalization is not the computational bottleneck in a typical QMC calculation, I changed norm_factor to use the exact expressions.

### CODE REVIEW

**Comment by rcclay** [comment]
ok to test

**Comment by jptowns** [comment]
I'm not sure I understand all the messages from the above builds, but it looks like complaints about AFQMC. Do I need to do something special in order to test my changes?

File: src/QMCHamiltonians/tests/test_paircorr.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+// File created by: Mark Dewing, markdewing@gmail.com, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "catch.hpp"
+#include "OhmmsData/Libxml2Doc.h"
+#include "Lattice/ParticleBConds.h"
+#include "Particle/ParticleSet.h"
+#include "Particle/DistanceTableData.h"
+#include "Particle/SymmetricDistanceTableData.h"
```
Comment (markdewing) [inline src/QMCHamiltonians/tests/test_paircorr.cpp]: This should only be included for AOS.

**Comment by markdewing** [comment]
This issue is included files for SOA vs. AOS.    Try removing the include of SymmetricDistanceTable.h.

File: src/QMCHamiltonians/tests/test_paircorr.cpp (hunk 1 above)
Comment (prckent) [reply src/QMCHamiltonians/tests/test_paircorr.cpp]: Maybe we should put AoS_only in their name?

File: src/QMCHamiltonians/tests/test_paircorr.cpp [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  elec->R[6][1] = 0.0;
+  elec->R[6][2] = 1.0;
+  elec->R[7][0] = 1.0;
+  elec->R[7][1] = 1.0;
+  elec->R[7][2] = 1.0;
+
+  elec->get(std::cout); // print particleset info to stdout
+
+  // Set up the distance table
+  const int ee_table_id = elec->addTable(*elec, DT_AOS, false);
```
Comment (markdewing) [inline src/QMCHamiltonians/tests/test_paircorr.cpp]: This should have an ifdef for SOA vs. AOS
```
#ifdef ENABLE_SOA
 ....  addTable(*eloc, DT_SOA, false);
#else
  ...  addTable(*eloc, DT_AOS, false);
#endif

Comment (jptowns) [reply src/QMCHamiltonians/tests/test_paircorr.cpp]: Wow. Turns out if you run in SoA mode, you can still create a AoS distance table. For the unit test, this meant that PairCorrEstimator basically never went down the SoA path in PairCorrEstimator::evaluate() !

Fixed and pushing new commit now.

**Pull request closed without merging**